lib/formats/gzip.js

```diff
@@ -97,52 +99,41 @@ function addTarEntry(pack, options, data = null) {
 }
 
 async function compressAttachments(apos, pack, attachments = {}) {
+
+  const copyOut = Promise.promisify(apos.attachment.uploadfs.copyOut);
+
   await addTarEntry(pack, {
     name: 'attachments/',
     type: 'directory'
   });
 
-  const chunkedPromises = chunkPromises(getPromises(), 5);
   let attachmentError = false;
 
-  for (const chunk of chunkedPromises) {
-    const results = await Promise.allSettled(chunk);
-    if (results.some(({ status }) => status === 'rejected')) {
-      attachmentError = true;
-    }
-  }
+  await Promise.map(Object.entries(attachments), processOneAttachment, {
+    concurrency: 5
```

Comment:
Keeps the pipeline full with at least 5 until done, rather than waiting for the last in each batch before going on